File: C++/Cristian Maldonado/Ordenamientos/Quick_Sort.cpp
#include<bits/stdc++.h>
#include<cstdlib>

using namespace std;

int partition (int arr [], int low, int high){
	int pivot = arr[high];
	int i = low - 1;
	for(int j = low; j < high; j++){
		if(arr[j] <= pivot){
			i++;
			int temp = arr[i];
			arr[i] = arr[j];
			arr[j] = temp;
		}
	}
	int temp = arr[i + 1];
	arr[i + 1] = arr[high];
	arr[high] = temp;
	return i + 1;
}
void quickSort(int arr[], int low, int high){
	if(low < high){
		int pivot = partition(arr, low, high);
		quickSort(arr, low, pivot - 1);
		quickSort(arr, pivot + 1, high);
	}
}

int main (int argc, char *argv[]) {
	string entrada;
	getline(cin, entrada);
	string intermediate;
    vector<int> vec;
    stringstream check1(entrada);
    while(getline(check1, intermediate, ' ')) {
	  vec.push_back(atoi(intermediate.c_str()));
    }
    int arr[vec.size()];
	for(int i = 0; i < vec.size(); i++){
        arr[i] = vec[i];
	}
	quickSort(arr, 0, sizeof(arr)/sizeof(arr[0])-1);
	for(int j = 0; j < vec.size(); j++){
		cout << arr[j] << " ";
	}
	return 0;
}

File: Codigos estudiantes por lenguaje/Java/Paula Rios/ordenamientos/quickSort.java
public class Main {
  static int partition(int arr[], int low, int high){
      int pivot=arr[high];
      int i=(low-1);
      for (int j=low; j <high; j++) {
          if(arr[j]<=pivot){
              i++;
              int temp=arr[i];
              arr[i]=arr[j];
              arr[j]=temp;
          }
      }
      //revisan el ultimo dato
      int temp=arr[i+1];
      arr[i+1]=arr[high];
      arr[high]=temp;
      return i+1;
  }
  
  static void quickSort(int arr[], int low, int high){
      if(low<high){
          int pi=partition(arr, low, high);
          quickSort(arr, low, pi-1);
          quickSort(arr, pi+1, high);
      }
  }
  static void print(int arr[]){
      int n=arr.length;
      for (int i = 0; i <n; i++) {
          System.out.print(arr[i]+" ");
      }
      System.out.println();
  }
    public static void main(String[] args) {
        int arr[]={10,7,8,9,1,5};
        print (arr);
        quickSort(arr, 0,arr.length-1);
        print(arr);
    }
}
